### PR TITLE
Set BuildingRef as output property from GetBuildVersion

### PR DESCRIPTION
**Pull request by aok-foss**
It's not a good idea to take a dependency on a branch name. At the same time, branches are too useful to ignore. The `PublicRelease` functionality depends on them.
The argument can go one step further: during development (in non-public releases), branch names in version numbers are useful for traceability: e.g. `1.0.0-g1234567+PR1512` or `1.0.0-g1234567+BalpineLinux`

It would be easiest to extend NBGV-produced version numbers in the way above if NBGV also provided access to build refs.

### CODE REVIEW

File: src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs [hunk 1]
```diff
@@ -20,11 +20,6 @@ public GetBuildVersion()
         {
         }
 
-        /// <summary>
-        /// Gets or sets the ref (branch or tag) being built.
-        /// </summary>
-        public string BuildingRef { get; set; }
```
Comment (AArnott) [inline src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs:26]: Removing this seems irrelevant, and would be a breaking change and drop support for an important use case.

Comment (aok-foss) [reply src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs:26]: Thought I was removing some dead code. Will restore.

File: src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs [hunk 2]
```diff
@@ -78,6 +78,13 @@ public GetBuildVersion()
         [Output]
         public bool PublicRelease { get; private set; }
 
+        /// <summary>
+        /// Gets the ref (branch or tag) being built.
+        /// Only specified when this is not a public release.
+        /// </summary>
+        [Output]
+        public string PrivateReleaseGitRef { get; set; }
```
Comment (AArnott) [inline src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs:86]: I don't see any benefit to only setting this when it's not a public release. We also have a `PublicRelease` output parameter so folks can condition their own code. Do you object to just making this `BuildingRef` and setting it unconditionally?

Comment (aok-foss) [reply src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs]: No objections. Updated.
